llvm/lib/IR/LLVMContextImpl.h

```diff
@@ -14,6 +14,7 @@
 #ifndef LLVM_LIB_IR_LLVMCONTEXTIMPL_H
 #define LLVM_LIB_IR_LLVMCONTEXTIMPL_H
 
+#include "AttributeImpl.h"
```

Comment:
Would a forward-declaration for ConstantRangeAttributeImpl be sufficient here?

Reply:
it compiles add check-llvm passes with forward-declaration also so seems like it :D